### PR TITLE
refactor(input-group): allow setting theme at runtime

### LINKED ISSUE #8619: Support changing theme dynamically on Input Group
## Problem
I need to change the theme in runtime using TaaS, but Inputs groups fail to update.

After changing my theme from Fluent to Material the labels still look as in fluent:
image

Related to this there's a problem. Even if I'm not switching themes, if the stylesheet hasn't loaded by the time the Input Group was rendered then it will default to material even after it finishes loading.
image

## Solution
I don't have a clear solution to this. Input Groups could observe some global theme property that I can update using a service.
I think this was @damyanpetev suggestion.

## Alternatives
I could destroy the view and force it to render from scratch so it infers the new theme.
This could be easily achieved with a fake *ngFor on an array with a single theme property.

### PR DESCRIPTION
**Pull request by simeonoff**
Closes #8619 

### Additional information (check all that apply):
 - [x] Bug fix
 - [x] New functionality
 - [x] Documentation
 - [ ] Demos
 - [ ] CI/CD

### Checklist:
 - [x] All relevant tags have been applied to this PR
 - [ ] This PR includes unit tests covering all the new code ([test guidelines](https://github.com/IgniteUI/igniteui-angular/wiki/Test-implementation-guidelines-for-Ignite-UI-for-Angular))
 - [x] This PR includes API docs for newly added methods/properties ([api docs guidelines](https://github.com/IgniteUI/igniteui-angular/wiki/Documentation-Guidelines))
 - [x] This PR includes `feature/README.MD` updates for the feature docs
 - [ ] This PR includes general feature table updates in the root `README.MD`
 - [x] This PR includes `CHANGELOG.MD` updates for newly added functionality
 - [ ] This PR contains breaking changes
 - [ ] This PR includes `ng update` migrations for the breaking changes ([migrations guidelines](https://github.com/IgniteUI/igniteui-angular/wiki/Update-Migrations))
 - [ ] This PR includes behavioral changes and the feature specification has been updated with them

### CODE REVIEW

**Review by rkaraivanov — DISMISSED** [review]

File: projects/igniteui-angular/src/lib/input-group/input-group.component.ts [hunk 1]
```diff
@@ -11,7 +11,7 @@ import {
     QueryList,
     Inject,
     Optional,
-    AfterContentInit
+    AfterContentInit,
```
Comment (kdinev) [inline projects/igniteui-angular/src/lib/input-group/input-group.component.ts:14]: I think the trailing comma here needs to go away.

**Review by kdinev — APPROVED** [review]
In general, I approve the change, but I think there will be a merge conflict with the ESLint PR, so I suggest it be pulled there, so for the PR to wait for a merge of the ESLint changes and changes to be resolved afterwards.

**Comment by simeonoff** [comment]
> In general, I approve the change, but I think there will be a merge conflict with the ESLint PR, so I suggest it be pulled there, so for the PR to wait for a merge of the ESLint changes and changes to be resolved afterwards.

Well, I want this to be released with 11.1, if the ESLint PR takes long to merge it doesn't make sense for me to have to wait for it.

**Comment by kdinev** [comment]
@simeonoff Let's merge it in master and then the conflict will be resolved with a merge to the eslint PR from master.

**Pull request merged**